lasagne/layers/normalization.py

```diff
@@ -101,3 +103,58 @@ def get_output_for(self, input, **kwargs):
             scale += self.alpha * input_sqr[:, i:i+ch, :, :]
         scale = scale ** self.beta
         return input / scale
+
+
+def standardize(layer, offset, scale, shared_axes):
+    """
+    Convenience function for standardizing inputs by applying a fixed offset
+    and scale.  This is usually useful when you want the input to your network
+    to, say, have zero mean and unit standard deviation over the feature
+    dimensions.  This layer allows you to compute those statistics over your
+    training set and include them as part of your network.  The statistics are
+    supplied as the `offset` and `scale` parameters, which are applied to the
+    input by subtracting `offset` and dividing by `scale`, sharing dimensions
+    as specified by the `shared_axes` argument.
+
+    Parameters
+    ----------
+    layer : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+    offset : Theano shared variable, expression, numpy array or callable
+        The offset to apply (via subtraction) to the axis/axes being
+        standardized.
+    scale : Theano shared variable, expression, numpy array or callable
+        The scale to apply (via division) to the axis/axes being standardized.
+    shared_axes : 'auto', int or tuple of int
+        The axis or axes which will be shared (i.e. are not being standardized
+        over).
```

Comment:
Comparing to the `BatchNormLayer`, those axes are actually the axes standardized over, not the axes *not* standardized over. I.e., to compute the offset for `shared_axes=(1,3)`, you would compute `np.mean(X, axis=(1,3))`. It's confusing. I think it may be safer to copy and adapt the explanation from `BiasLayer` or `ScaleLayer`.